common/gps.h

```diff
@@ -0,0 +1,99 @@
+//GPS
+
+
+#include "NMEA_parser.h"
+#include "serial_reader.h"
+#include "../third_party/units.h"
+#include "map_coordinate.h"
+
+
+
+
+class Gps {
+    using degree_t = units::angle::degree_t;
+    using meter_t = units::length::meter_t;
+    using meters_per_second_t = units::velocity::meters_per_second_t;
+    private:
+    
+    char *m_device_path;
+    int m_currentNumberOfSats;
+    int m_gpsQuality;
+    double m_horizontalDilution;
+    meter_t m_altitude;
+    degree_t m_longitude;
+    degree_t m_latitude;
+    meters_per_second_t m_velocity;
+
+    bool getGPSPosition() {
+        using namespace units::angle;
+        using namespace units::literals;
+
+        arcminute_t latmin;
+        arcminute_t longmin;
+        degree_t lat;
+        degree_t longitude;
+        std::string latDir;
+        std::string longDir;
+
+        // reading and parsing serial data
+        std::string serialString = Serial_reader::readSerialUSB(m_device_path);
```

Comment:
You shouldn't re-open the serial port every time you read a line. Could you try cherry picking 23cf3e0c8c218f18569b0d78b73c6c31395594d8 which intergrates the libserial library? Then all you need to do is open it and set the baud rate in the constructor: 
```c++
Gps::Gps(const std::string &device) : m_SerialPort(device)
{
    m_SerialPort.Open(SerialPort::BAUD_9600);
}
```
and call a single method to get lines:
```c++
const std::string serialString = m_SerialPort.ReadLine(5000) 
```

Reply:
:+1: